mobile_verifier/src/reward_shares.rs

```diff
@@ -183,9 +288,90 @@ mod test {
 
+    #[tokio::test]
+    async fn transfer_reward_amount() {
+        let owner: PublicKeyBinary = "112NqN2WWMwtK29PMzRby62fDydBJfsCLkCAf392stdok48ovNT6"
+            .parse()
+            .expect("failed owner parse");
+        let payer: PublicKeyBinary = "11sctWiP9r5wDJVuDe1Th4XSL2vaawaLLSQF8f8iokAoMAJHxqp"
+            .parse()
+            .expect("failed payer parse");
+
+        let data_transfer_sessions = stream::iter(vec![ValidDataTransferSession {
+            pub_key: owner.clone(),
+            payer,
+            upload_bytes: 66,
+            download_bytes: 1,
+            num_dcs: 0, // Not used
+            first_timestamp: DateTime::default(),
+            last_timestamp: DateTime::default(),
+        }]);
+
+        let now = Utc::now();
+        let epoch = (now - Duration::hours(1))..now;
+
+        let data_transfer_rewards =
+            TransferRewards::from_transfer_sessions(dec!(1.0), data_transfer_sessions, &epoch)
+                .await
+                .expect("Could not fetch data transfer sessions");
+
+        assert_eq!(data_transfer_rewards.reward(&owner), dec!(0.00000006));
+        assert_eq!(data_transfer_rewards.scale, dec!(1.0));
+        // Is this correct?
+        assert_eq!(
+            data_transfer_rewards.remaining_rewards,
+            dec!(4166666666666.6666666066666666)
+        );
+    }
+
+    #[tokio::test]
+    async fn transfer_reward_scale() {
+        let owner: PublicKeyBinary = "112NqN2WWMwtK29PMzRby62fDydBJfsCLkCAf392stdok48ovNT6"
+            .parse()
+            .expect("failed owner parse");
+        let payer: PublicKeyBinary = "11sctWiP9r5wDJVuDe1Th4XSL2vaawaLLSQF8f8iokAoMAJHxqp"
+            .parse()
+            .expect("failed payer parse");
+
+        // Just an absurdly large amount of DC
+        let mut transfer_sessions = Vec::new();
+        for _ in 0..1_000_000 {
+            transfer_sessions.push(ValidDataTransferSession {
+                pub_key: owner.clone(),
+                payer: payer.clone(),
+                upload_bytes: 66 * 80_000_000 * 95_000_000,
+                download_bytes: 0,
+                num_dcs: 0, // Not used
+                first_timestamp: DateTime::default(),
+                last_timestamp: DateTime::default(),
+            });
+        }
+
+        let data_transfer_sessions = stream::iter(transfer_sessions);
+
+        let now = Utc::now();
+        let epoch = (now - Duration::hours(24))..now;
+
+        let data_transfer_rewards =
+            TransferRewards::from_transfer_sessions(dec!(1.0), data_transfer_sessions, &epoch)
+                .await
+                .expect("Could not fetch data transfer sessions");
+
+        assert_eq!(
+            data_transfer_rewards.remaining_rewards,
+            dec!(20_000_000) * dec!(1_000_000)
+        );
+        assert_eq!(
+            // Rewards are automatically scaled
+            data_transfer_rewards.reward(&owner),
+            dec!(80_000_000) * dec!(1_000_000)
+        );
```

Comment:
be good to assert the expected scale value here

Reply:
I tried to make it as close to `0.5` to do just that